test-js/unit/pentaho/type/simple.serialization.Spec.js

```diff
@@ -0,0 +1,259 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "pentaho/type/Context"
+], function(Context) {
+  "use strict";
+
+  /*global describe:false, it:false, expect:false, beforeEach:false, Date:false */
+
+  describe("instance serialization -", function() {
+    var context = new Context();
+
+    describe("pentaho/type/simple -", function() {
+
+      // Using pentaho/type/boolean because pentaho/type/simple is abstract
+      var PentahoBoolean = context.get("pentaho/type/boolean");
+      var originalSpec = {v: false, f: "I'm a simple value"};
+
+      describe("values", function() {
+
+        describe("value with formatted info", function() {
+          var value;
+
+          beforeEach(function() {
+            value = new PentahoBoolean(originalSpec);
+          });
+
+          describe("default", function() {
+            it("should return primitive value, formatted value, and inline type", function() {
+              var spec = value.toSpec();
+
+              expect(spec.v).toBe(originalSpec.v);
+              expect(spec.f).toBe(originalSpec.f);
+              expect(spec._).toBeDefined();
+              expect(spec._).toBe(PentahoBoolean.type.toSpec().id);
```

Comment:
Don't like it as well. In a way, we want to avoid testing the `Type#toReference()` and `Type#toSpec()` which are part of another story, ... The problem is that the id is really "pentaho/type/boolean", so imagine that this for was used to escape that. This test will break as soon the the other story kicks in, as we'll really start returning the short id, "boolean"...
I'll change it to `toEqual(Pentaho.type.toReference())` instead and it will work then as well.